src/display/editor/tools.js

```diff
@@ -42,6 +42,78 @@ function bindEvents(obj, element, names) {
   }
 }
 
+/**
+ * Class to store current pointer used by the editor.
+ */
+class CurrentPointer {
+  // To manage the pointer events.
+  static #pointerId = NaN;
+
+  static #pointerType = null;
+
+  static #pointerIds = null;
+
+  static #moveTimestamp = NaN;
+
+  static initializeAndAddPointerId(pointerId) {
+    (CurrentPointer.#pointerIds ||= new Set()).add(pointerId);
+  }
+
+  static setPointer(pointerType, pointerId) {
+    CurrentPointer.#pointerId ||= pointerId;
+    CurrentPointer.#pointerType ??= pointerType;
+  }
+
+  static setTimeStamp(timeStamp) {
+    CurrentPointer.#moveTimestamp = timeStamp;
+  }
+
+  static isSamePointerId(pointerId) {
+    if (CurrentPointer.#pointerId === pointerId) {
+      return true;
+    }
+
+    CurrentPointer.#pointerIds?.delete(pointerId);
+    return false;
+  }
+
+  static isPointerType() {
```

Comment:
I'm not an expert in english, but the name looks strange for me... I'd have said `hasPointerType`.
I just read again https://github.com/mozilla/pdf.js/pull/20213/files#diff-02919db67aef2f378464ba4c97e5f59e1e114498a147113025cd4163287f25f4R670 and I've the feeling that this class should be initialized with this the pointer type and the static methods in this class shouldn't be static. An instance could be created in startDrawing and set in a static member of DrawingEditor.

Reply:
I've changed the name.
For the logic, I feel that we need a static class outside of the editor if we want all editors to share the same pointer type. It's really important for pdf with multiple pages as each page has its own editor, and you don't want to reinitialized the pointer type on each page. Furthermore, when you use this mode for the first time, you want to initialize the pointer type, and the only way to do that is to check if the pointer type is null. Also, you will have to put the class with a getter/setter if you want to be able to clear it outside of the DrawingEditor (it's the only way to check if the mode is changed to the best of my knowledge, or perhaps through ``disableEditing``).

Reply:
Thanks for the answer. Even if you are okay with that, I want to highlight the fact that I don't think it's just an eraser issue. Imagine that you are drawing on one page with a stylus, then you scroll with your finger to the next page, you use your finger to stop the scroll, and finally you start drawing: I'm not convinced that the pointer type will be set on stylus and not touch when you try drawing.